databases/week3/lesson3.sql

```diff
@@ -0,0 +1,231 @@
+--create the database and queries for a meal sharing website
+CREATE database meal_website;
+use meal_website;
+SET NAMES utf8mb4;
+
+--Db model has 3 tables
+ 
+CREATE TABLE `meal` (
+    id INT(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+    title VARCHAR(255),
+    description TEXT ,
+    location VARCHAR(255),
+    `when` DATETIME,
+    max_reservations INT(10),
+    price DECIMAL,
+    created_date DATE 
+);
+
+CREATE TABLE `reservation` (
+    id INT(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+    number_of_guests INT (10),
+    meal_id INT (10) UNSIGNED NOT NULL,
+    created_date DATE,
+    contact_phonenumber VARCHAR (255),
+    contact_name VARCHAR (255),
+    contact_email VARCHAR (255),
+
+    FOREIGN KEY (meal_id) REFERENCES meal (id) 
+    ON DELETE CASCADE 
+    ON UPDATE CASCADE
+);
+
+CREATE TABLE `review` (
+    id INT(10) UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+    title VARCHAR (255),
+    description TEXT,
+    meal_id INT(10) UNSIGNED NOT NULL,
+    stars INT (10),
+    created_date DATE,
+
+    FOREIGN KEY (meal_id) REFERENCES meal (id) 
+    ON DELETE CASCADE 
+    ON UPDATE CASCADE
+);
+
+-------------------queries-----------------
+------------ Meal---
+--Get all meals
+SELECT *
+FROM meal;
+
+-- Add a new meal
+INSERT INTO meal VALUES (1,'homemade kebab', 'lamb and calf meat', 'Tehran', '2021-08-08', 10,200, '2021-08-07');
+INSERT INTO meal (
+    title,
+    description,
+    location,
+    `when`,
```

Comment:
Hi @zmzade ,
When you are writing queries make sûre to follow just one way to do that: When you are writing `INSERT INTO meal (
    title,
    description,
    location,
    `when`,...)`, choose if you will write all of them with backticks or without it. It's actually not a big problem but it's just a matter of following some best practices.

Reply:
Yes you are right. But there was a problem, the code did not make the table, and I could not find anything wrong in the code. Error was something from 'when', I changed it to backtick and then the table created. :)